lib/puppet/type/selinux_port.rb

```diff
@@ -0,0 +1,48 @@
+Puppet::Type.newtype(:selinux_port) do
+  @doc = 'Manage SELinux port definitions. You should use selinux::port instead of this directly.'
+
+  ensurable
+
+  newparam(:title, namevar: true) do
+    desc 'Should be of the form "protocol_port" or the type may misbehave'
+  end
+
+  newproperty(:ports) do
+    desc 'The port or port range to manage'
+    isrequired
+
+    validate do |value|
+      vals = value.split('-')
+      debug(vals)
+      vals.each do |val|
+        val = Integer(val)
+        if val < 1 || val > 65_535
+          raise ArgumentError, "Illegal port value '#{val}'"
+        end
+      end
+
+      if (vals.count != 1) && (Integer(vals[0]) >= Integer(vals[1]))
+        raise ArgumentError, "Invalid port range #{value}"
+      end
+    end
+  end
+
+  newproperty(:protocol) do
+    desc 'The protocol of the SELinux port definition'
+    newvalues(:tcp, :udp, :ipv4, :ipv6)
+    isrequired
+  end
+
+  newproperty(:context) do
```

Comment:
I think this should be renamed. context is the combination of `$selinux-user:$selinux-role:$selinux-type:$selinux-range` whereas this only is the `$selinux-type` part. 

the file resource calls the parameters `seuser`, `setype` , 'se...` maybe we should use the naming too?